UPGRADE.md

```diff
@@ -23,6 +23,11 @@ The only users who need to take action are those who are explicitly relying on t
 
 We encourage all users to review their configuration, but we do not anticipate that this change will require any action for most users.
 
+### Default resources are not created when creating a new `Mesh`
+
+We decided to remove default `TrafficRoute` and `TrafficPermission` policies that were created during a new mesh creation. Since this release your applications can communicate without need to apply any policy by default.
+If you want to keep the previous behaviour set `KUMA_DEFAULTS_CREATE_MESH_ROUTING_RESOURCES` to `true`.
```

Comment:
I believe this needs to be enhanced as we should explain in more details. Changes included in this PR are not trivial. To better inform our users about them, we should provide a comprehensive explanation of the following:

* **Policies no longer created by default**: Clearly outline the specific policies that are no longer being automatically installed.

* **Policies created by default**: Clearly identify the new policies that will be automatically installed.

* **Scenarios requiring `KUMA_DEFAULTS_CREATE_MESH_ROUTING_RESOURCES`**: Provide detailed examples and scenarios where setting the `KUMA_DEFAULTS_CREATE_MESH_ROUTING_RESOURCES` flag is necessary. This will help users understand the specific cases where manual intervention is required.

* **Migration steps and considerations**: Outline potential migration steps or considerations for users who are transitioning from the previous policy behavior. This will help them navigate the changes smoothly and ensure their deployments remain stable and secure.

This enhanced version provides a more detailed and structured explanation of the changes, making it easier for users to understand the implications and take the necessary actions.

Reply:
We can work on it in the stabilization week though if it were to block this PR

Reply:
Added some notes, missing `Migration steps and considerations`